api/src/main/java/io/opencensus/trace/Span.java

```diff
@@ -244,6 +257,22 @@ public final void end() {
     end(EndSpanOptions.DEFAULT);
   }
 
+  /**
+   * Marks the end of {@code Span} execution with the given options and end time.
+   *
+   * <p>Only the timing of the first end call for a given {@code Span} will be recorded, and
+   * implementations are free to ignore all further calls. The default implementation ignores
+   * endTimestamp parameter and calls {@link end} method. Implementations that keep track of end
+   * time are expected to override this method.
+   *
+   * @param options the options to be used for the end of the {@code Span}.
+   * @param endTimestamp the {@link Timestamp} to set end time of the span to.
+   * @since 0.23
+   */
+  public void end(EndSpanOptions options, Timestamp endTimestamp) {
```

Comment:
can endTimestamp be part of EndSpanOptions?

Reply:
Interesting, I like the suggestion. I can try that, but since Bogdan has a different idea I will wait for now.